### PR TITLE
UserInformation

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have solved third task: create three tables: users, contact, address. Created separate methods to display information from each table.

### CODE REVIEW

File: src/com/clouway/users/adapter/jdbc/ConnectionProvider.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  public Connection provide() {
+    try {
+      return connection = DriverManager.getConnection(database, user, pass);
+    } catch (SQLException e) {
+    }
+    return null;
+  }
+
+  @Override
+  public void close() {
```
Comment (freekman) [inline src/com/clouway/users/adapter/jdbc/ConnectionProvider.java]: .

File: src/com/clouway/users/core/Provider.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+   *
+   * @return resource
+   * @throws SQLException
+   */
+  T provide();
+
+  /**
+   * Will close resource
+   */
+  void close();
```
Comment (freekman) [inline src/com/clouway/users/core/Provider.java]: .

Comment (Stanislava88) [inline src/com/clouway/users/core/Provider.java]: fixed

File: src/com/clouway/users/core/UserContactRepository.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+   */
+  void register(Contact contact);
+
+  /**
+   * Will return contact
+   *
+   * @param id id for this contact
+   * @return contact registered for this id
+   */
+  Contact findById(int id);
```
Comment (freekman) [inline src/com/clouway/users/core/UserContactRepository.java]: if i am an user and i Use ```findById``` i would expect to see an Contact object containing info for User + Address not some ID's i thought we have discused this

File: src/com/clouway/users/core/UserContactRepository.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * The implementation of this interface will be used to save and retrieve contacts
+ *
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public interface UserContactRepository {
+  /**
+   * @param contact registered contact
+   */
+  void register(Contact contact);
```
Comment (freekman) [inline src/com/clouway/users/core/UserContactRepository.java]: do you really need this register

**Comment by freekman** [comment]
LGTM

**Pull request merged**